swap_meet/vendor.py

```diff
@@ -1,2 +1,39 @@
-class Vendor:
-    pass
\ No newline at end of file
+class Vendor: 
+    def __init__(self,inventory = None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item 
+
+    def remove(self,item):
+        if item not in self.inventory:
+            return False 
+        self.inventory.remove(item)
+        return item 
+
+    def get_by_category(self,category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:  
+            friend.inventory.append(my_item)
+            self.inventory.append(their_item)
+            friend.inventory.remove(their_item) 
+            self.inventory.remove(my_item)
```

Comment:
This works! I also see we created helper methods above, the `add()` and `remove()` seem useful...